### ISSUE TITLE
no-uninitialized: False positive in ambient context, in IntelliJ IDEA

### ISSUE THREAD

**Issue by UselessPickles** [palantir/tslint#3570]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.2
- __Running TSLint via__: IntelliJ IDEA

#### TypeScript code being linted

File name : "json-module.d.ts"
```ts
// Allows json files to be imported as a module
declare module "*.json" {
    import {JSONObject} from "src/json";
    const value: JSONObject;
    export default value;
}
```

Contents of "src/json.ts" for reference (although, it's likely irrelevant)
```ts
/**
 * Type declarations and utilities for working with JSON-compatible data structures.
 * @module
 */

/**
 * Any primitive data type that is allowed in JSON.
 */
export type JSONPrimitive = string | number | boolean | null;

/**
 * Any value data type allowed in JSON (primitives, arrays, and objects).
 */
export type JSONValue = JSONPrimitive | JSONArray | JSONObject;

/**
 * An array of data types allowed in JSON.
 */
export interface JSONArray extends Array<JSONValue> {
}

/**
 * An object whose properties are of types allowed in JSON.
 */
export interface JSONObject {
    [name: string]: JSONValue;
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:recommended",
    "rulesDirectory": [
        "node_modules/tslint-strict-null-checks/rules"
    ],
    "rules": {
        "max-line-length": {
            "options": [120],
            "severity": "warn"
        },
        "typedef": [
            true,
            "call-signature",
            "parameter",
            "property-declaration"
        ],
        "new-parens": true,
        "no-var-requires": false,
        "no-arg": true,
        "no-conditional-assignment": true,
        "no-console": [
            true,
            "debug",
            "info",
            "log",
            "time",
            "timeEnd",
            "trace"
        ],
        "trailing-comma": [
            true,
            {
                "multiline": "never",
                "singleline": "never"
            }
        ],
        "object-literal-sort-keys": false,
        "array-type": [true, "array"],
        "ordered-imports": false,
        "quotemark": [true, "double", "jsx-double", "avoid-escape"],
        "class-name": false,
        "interface-name": false,
        "no-namespace": false,
        "no-default-export": true,
        "no-unused-variable": true,
        "object-literal-shorthand": false,
        "object-literal-key-quotes": [true, "consistent"],
        "max-classes-per-file": [false],
        "triple-equals": [true, "allow-null-check"],
        "no-uninitialized": [true, "variables", "properties"],
        "no-empty-interface": false,
        "no-string-literal": false,
        "jsdoc-format": true,
        "member-ordering": false,
        "forin": true,
        "no-for-in-array": true,
        "no-invalid-template-strings": true,
        "semicolon": [true, "always", "strict-bound-class-methods"]
    },
    "jsRules": {
        "max-line-length": {
            "options": [120],
            "severity": "warn"
        }
    }
}
```

#### Actual behavior

IntelliJ IDEA complains about `value` not being initialized, via tslint rule "no-uninitialized".

<img width="613" alt="screen shot 2017-12-11 at 5 13 07 pm" src="https://user-images.githubusercontent.com/22800095/33857021-0a317bf2-de98-11e7-953a-c0bf63958e94.png">

However, running tslint from the command line on the same project does NOT complain about this.

#### Expected behavior

The "no-uninitialized" rule should not be applied in "ambient contexts" like module declarations in ".d.ts" files.

**Comment by UselessPickles** [comment]
I suspect this may be a bug that I should be reporting to IntelliJ, but I thought I'd start here first just in case. If this is not a tslint bug directly, please point me in the right direction of where I should be reporting this bug.

**Comment by ajafff** [comment]
`no-uninitialized` is not a TSLint core rule. You should bug the author of `tslint-strict-null-checks`

**Comment by UselessPickles** [comment]
Oops. The mismatch between the rule name "no-uninitialized" and the 3rd party plugin project name "tslint-strict-null-checks" threw me off.

FWIW, here's the issue reported on the correct project: https://github.com/alhugone/tslint-strict-null-checks/issues/6
